src/test/scala/com/exini/dicom/data/ByteParserTest.scala

```diff
@@ -0,0 +1,92 @@
+package com.exini.dicom.data
+
+import akka.util.ByteString
+import com.exini.dicom.data.ByteParser._
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.collection.mutable.ArrayBuffer
+
+class ByteParserTest extends AnyFlatSpec with Matchers {
+
+  import ByteParserTest._
+
+  "Parsing a string of bytes" should "produce the correct values for valid byte sequence" in new Fixture(
+    Seq(ByteString("STARTBoat,Car,Airplane"))
+  ) {
+    parse()
+    result shouldBe Seq("Boat", "Car", "Airplane")
+    isCompleted shouldBe true
+  }
+
+  it should "throw an error for invalid byte sequence" in new Fixture(Seq(ByteString("Car,Boat"))) {
+    assertThrows[DicomParseException] {
+      parse()
+    }
+  }
+
+  it should "accept data in chunks" in new Fixture(
+    Seq(ByteString("STARTBoat,Car"), ByteString("Airplane"))
```

Comment:
Shouldn't that be `,Airplane` (with comma separating `Car` and `Airplane`)?

Reply:
In this particular parser, words can be separated into chunks, or within a chunk by a comma.